#3535 Renault: Provide a meaningful error when not paired

ickeundso opened the pull request:
- log.ERROR.Println added to kamereonVehicles
- no logic changes only a log messages is added

It would also be possible to move this to the ==ACTIVE branch. But I don't know how the kamereon API will behave with other versions

ickeundso committed:
#3535 Renault: Provide a meaningful error when not paired

- log.ERROR.Println added to kamereonVehicles
- no logic changes only a log messages is added

It would also be possible to move this to the ==ACTIVE branch. But I don't know how the kamereon API will behave with other versions

ickeundso commented on vehicle/renault.go:
```diff
@@ -318,6 +324,11 @@ func (v *Renault) kamereonVehicles(accountID string) ([]string, error) {
 			if strings.ToUpper(v.Status) == "ACTIVE" {
 				vehicles = append(vehicles, v.VIN)
 			}
+			if v.ConnectedDriver.Role != "MAIN_DRIVER" {
```
Yes, you are right. I just wondered if there are other version of the API that maybe do not have this property so I would just start with a logging an error.  But yes, we could also continue here with the more strict implementation for evcc.

andig replied:
I‘d more ask if secondary drivers or others might work too? That would need to be validated.

ickeundso committed:
Renault: Provide a meaningful error when not paired

- do not only log the not paired vehicle but create an error response the missconfigured / non-paired vehicle

ickeundso committed:
Merge remote-tracking branch 'origin/master' into #3535

ickeundso committed:
Merge remote-tracking branch 'origin/master' into #3535

ickeundso committed:
Renault: Provide a meaningful error when not paired

- do not only log the not paired vehicle but create an error response the missconfigured / non-paired vehicle
- do not check for role "MAIN_DRIVER" string, just check of role existence. The "ConnectedDriver.Role" is not documented at https://renault-api.readthedocs.io/en/latest/reference/kamereon.html#renault_api.kamereon.models.KamereonVehiclesLink but I can reproduce the behaviour. For the case the vehicle is paired the ConnectedDriver.Role attribute is there an has the "MAIN_DRIVER" value and if the vihicle is not paired the ConnectedDriver.Role attribute is not there, the API can be tested of empty string.

andig commented on vehicle/renault.go:
```diff
@@ -147,7 +152,9 @@ func NewRenaultFromConfig(other map[string]interface{}) (api.Vehicle, error) {
 
 	if err == nil {
 		v.vin, err = ensureVehicle(cc.VIN, func() ([]string, error) {
-			return v.kamereonVehicles(v.accountID)
```
Das ist unnötig

andig commented on vehicle/renault.go:
```diff
@@ -311,17 +318,30 @@ func (v *Renault) kamereonPerson(personID string) (string, error) {
 	if err == nil {
 		for _, v := range res.VehicleLinks {
 			if strings.ToUpper(v.Status) == "ACTIVE" {
-				vehicles = append(vehicles, v.VIN)
```
Das geht in die falsche Richtung. Entweder geben wir nur gepairte Fahrzeuge zurück, oder wir geben z.B. `Vehicle` Objekte zurück so dass die aufrufende Funktion sich den Pairingstatus anschauen kann.

Ist sicher, dass `ConnectedDriver.Role` das relevante Setting ist?

ickeundso committed:
Renault: Provide a meaningful error when not paired

- remove not needed local var

andig commented:
@ickeundso die Validierung ist an der falschen Stelle. Vorschlag:
- alle Autos als Objekt zurück geben
- Validierung NUR machen falls das Auto auch ausgewählt ist

Willst Du das noch ändern? Sonst würde ich hier zu machen.

ickeundso commented:
ich schaue mir das nochmal an das man Fahrzeuge auswählen kann war mir nicht bewußt ..

ickeundso commented:
Auch ich glaube jetzt verstehe ich, die menge der Fahrzeuge die im Renault-Account gefunden werden muss ja nicht mit denen im evcc konfigurierten übereinstimmen. Und nur die interessieren uns ja hier.

andig commented:
Ich machs mal zu- wir können den Faden gerne wieder aufnehmen wenns soweit ist!
